### PR TITLE
Use TS_XXX_CTX_set_flags for TS RESP and VERIFY

### PR DESCRIPTION
**Pull request by richsalz**
The undocumented function TS_VERIFY_CTX_set_flags() has been
modified to "or in" the flag bits, like the other nearly
two-dozen xxx_set_flags() functions. Developers who want the
previous behavior should call the new function
TS_VERIFY_CTX_clear_flags() first.

-----

*OF COURSE* the TS code is different from the rest of the project.  This changes the semantics of an obscure undocumented function so that the behavior is consistent.  Not a big deal if you don't want this.  I came across it via #12666 when I was thinking of suggesting "add_flags" as a better API and it turned out "set_flags" was already used many many times, with the "or it in" semantics.  (I checked most of them.)

### CODE REVIEW

**Comment by slontis** [comment]
I don't think we should be doing this for 3.0 since it is a change in behavior.
You added these in 2015 :) so they have been around for a while..

File: CHANGES.md [hunk 1]
```diff
@@ -23,6 +23,14 @@ OpenSSL 3.0
 
 ### Changes between 1.1.1 and 3.0 [xx XXX xxxx]
 
+ * Deprecate TS_VERIFY_CTX_add_flags and TS_RESP_CTX_add_flags have been
```
Comment (slontis) [inline CHANGES.md:26]: If someone thinks this should be merged then you would need to fix up this first sentence.

Comment (richsalz) [reply CHANGES.md:26]: if it gets one approval i will fix the sentence else why bother.

**Comment by t-j-h** [comment]
Thanks for the prompt on the inconsistency. I did a quick check of the usage and it is rather mixed.
I don't think that we should change things - but the issue it raises should be discussed as the semantic difference is unfortunate.

These APIs are set-is-or:

BIO_set_flags
BN_set_flags
DH_set_flags
DSA_set_flags
EC_KEY_set_flags
EVP_MD_CTX_set_flags
EVP_CIPHER_CTX_set_flags
HMAC_CTX_set_flags
RSA_set_flags
X509_STORE_CTX_set_flags
X509_VERIFY_PARAM_set_flags
SSL_CONF_CTX_set_flags
SSL_CTX_dane_set_flags
SSL_dane_set_flags

These APIs are set-is-set:

ASN1_PCTX_set_flags
BN_BLINDING_set_flags
CONF_imodule_set_flags
DH_meth_set_flags
DSA_meth_set_flags
ENGINE_set_flags
EVP_CIPHER_meth_set_flags
evp_encode_ctx_set_flags
EVP_MD_meth_set_flags
ffc_params_set_flags
WPACKET_set_flags
RSA_meth_set_flags
TS_VERIFY_CTX_set_flags

**Comment by richsalz** [comment]
You did a much more exhaustive review than I did.  Impressive.  Closing this.

**Comment by slontis** [comment]
Could you maybe open an issue.. that puts Tim info in it.. It should probably be addressed in a later version (just not for 3.0).
Even with a  link to your closed PR maybe?

**Comment by richsalz** [comment]
Created an issue with a couple of links here, https://github.com/openssl/openssl/issues/12804

**Comment by slontis** [comment]
Thanks Rich and Tim.

**Pull request closed without merging**